### DISCUSSION TITLE
Selling Partner API for Direct Fulfillment Orders

### DISCUSSION THREAD

**Discussion by Billal-Munir** [amzn/selling-partner-api-models#3929] [General]
Hi, 

I am trying to invoke the Direct Fulfilment Orders API for the operations but not able to call
- **getOrders** (https://sandbox.sellingpartnerapi-na.amazon.com/vendor/directFulfillment/orders/v1/purchaseOrders?createdAfter=XXXX&MarketplaceIds=XXXX)
- **submitAcknowledgement** (https://sandbox.sellingpartnerapi-na.amazon.com/vendor/directFulfillment/orders/v1/acknowledgements)
 
when i done an investigation it came into my consideration that as it is restricted operation so need to get RDT from Tokens API using LWA and AWS STS Token. 

I tried to invoke the tokens API in sandbox with the below POST request in body and LWA token in header and STS token in Authorization but getting below mentioned error.

Request Body:

{
    "restrictedResources": [
        {
            "method": "GET",
            "path": "vendor/directFulfillment/orders/v1/purchaseOrders"
        }
    ],
    "targetApplication": "amz1.sellerapps.app.target-application"
}

Response Body:
{
    "errors": [
        {
            "code": "InvalidInput",
            "message": "Could not match input arguments",
            "details": ""
        }
    ]
}

Can some please help me here for below points:

1- what mistake I am doing here and why it is giving me an error
2- I have configured IAM User, IAM Role, IAM Policy and defined the Developer Profile and created app and self-authorized it with the seller account. Is it ok or do i need to with some other account type (Vendor Central).
3- Further to get RDT from Token API for Direct Fulfillment Purchase order API the request body mentioned above is ok or what is wrong with this request.
4- Can someone please guide me to get the correct request payload for POST method in Token API to get RDT. I have gone through documentation and issue posted on community but unable to find the correct one. 

Your swift response and help in this regards will be highly appreciated. Thanks.

Regards,
Bilal Ahmed Munir

**Comment by ritikacd** [comment]
Hi @Billal-Munir 
Vendor Direct Fulfillment APIs are only available for Vendors, which means you need to get an authorization from a Vendor Central account to access those APIs. More details here on the API models available for Sellers and Vendors: https://developer-docs.amazon.com/sp-api/docs/sp-api-models 

If you get an authorization from a Vendor, you can use an LWA access token to call Sandbox endpoints. You only need to create an RDT to call production Direct Fulfillment API endpoints. 

Thanks,
Ritika

**Comment by hubinjie** [comment]
may be you "path": "vendor/directFulfillment/orders/v1/purchaseOrders" change to "path": "/vendor/directFulfillment/orders/v1/purchaseOrders" will be works

**Comment by summanbahadurVolo** [comment]
Hi , i am also facing a simillar issue,
I am not able to generate RDT token even for production API

request body:
{ "restrictedResources":
    [
        { "method": "GET" , "path": "/vendor/directFulfillment/orders/v1/purchaseOrders"}
    ] 
}


response:
{
    "errors": [
        {
            "code": "InvalidInput",
            "message": "Application do not have access to some or all requested resource",
            "details": ""
        }
    ]
}

Note: if i add targetApplication with client application id , it does generates a token for me but that again does not helps while calling the API.

please suggest what could be wrong
